frontend/src/app/settings/change-username/change-username.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { FormControl, FormGroup } from '@angular/forms';
+import { AuthService } from '@services';
+import { formIcons } from '@shared/icons';
+import { map } from 'rxjs';
+
+@Component({
+    selector: 'ia-change-username',
+    templateUrl: './change-username.component.html',
+    styleUrl: './change-username.component.scss'
+})
+export class ChangeUsernameComponent {
+    currentName$ = this.authService.currentUser$.pipe(
+        map(user => user.name)
+    );
+
+    form = new FormGroup({
+        name: new FormControl<string>(''),
+    });
+
+    formIcons = formIcons;
+
+    invalidFeedback: string;
+    successFeedback: string;
+    errorFeedback: string;
+
+    constructor(
+        private authService: AuthService,
+    ) {
+        this.authService.currentUser$.pipe(
+            map(user=> user.name),
+            takeUntilDestroyed(),
+        ).subscribe(name => this.form.setValue({name}))
+    }
+
+    submitName() {
+        this.invalidFeedback = undefined;
+        this.successFeedback = undefined;
+        this.errorFeedback = undefined;
+
+        this.authService.updateSettings(this.form.value).subscribe({
+            next: this.onSuccess.bind(this),
+            error: this.onRequestFail.bind(this),
```

Comment:
Won't `next: this.onSuccess()` work?

Reply:
That would be `next: this.onSuccess` or `next: () => this.onSucces()`. (You provide the function, not the outcome.)

The first one (without bind) would not work because `onSuccess` is a bound function (i.e. it references `this`). You could do the second. I'm 50/50 on which is preferable.